plugins/modules/sssd_info.py

```diff
@@ -0,0 +1,230 @@
+            - V(list_servers) - List all servers for domain.
+        type: str
+        required: true
+        choices: ['domain_status', 'domain_list', 'active_servers', 'list_servers']
+    domain:
+        description:
+            - Domain name to check.
+            - Required for O(action=domain_status), O(action=active_servers) and O(action=list_servers).
+        type: str
+    server_type:
+        description:
+            - The type of server to retrieve for the O(action=active_servers) and O(action=list_servers).
+        type: str
+        choices: ['IPA', 'AD']
+extends_documentation_fragment:
+    - community.general.attributes
+'''
+
+EXAMPLES = r'''
+- name: Check SSSD domain status
+  community.general.sssd_info:
+    action: domain_status
+    domain: example.com
+  register: sssd_status_result
+
+- name: Get domain list
+  community.general.sssd_info:
+    action: domain_list
+  register: domain_list_result
+
+- name: Get active IPA servers for a domain
+  community.general.sssd_info:
+    action: active_servers
+    domain: example.com
+    server_type: IPA
+  register: active_servers_result
+
+- name: List servers for a domain
+  community.general.sssd_info:
+    action: list_servers
+    domain: example.com
+    server_type: AD
+  register: list_servers_result
+'''
+
+RETURN = r'''
+online:
+    description: The online status of the SSSD domain.
+    type: str
+    returned: when O(action=domain_status)
+    sample: online
+domain_list:
+    description: List of SSSD domains.
+    type: list
+    returned: when O(action=domain_list)
+    sample: ["ipa.domain", "winad.test"]
+servers:
+    description: Active servers for the specified domain and type.
+    type: dict
+    returned: when O(action=active_servers)
+    sample: {
+        "Global Catalog": "server1.winad.test",
+        "Domain Server": "server2.winad.test"
+    }
+list_servers:
+    description: List of servers for the specified domain.
+    type: list
+    returned: when O(action=list_servers)
+    sample: ["server1.winad.test", "server2.winad.test"]
+error:
+    description: Error message if operation failed.
+    type: str
+    returned: on error
+    sample: "Domain not found: winad.test"
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+import dbus
+
+
+class SSSDHandler:
+    """SSSD D-Bus handler"""
+    
+    BUS_NAME = 'org.freedesktop.sssd.infopipe'
+    DOMAIN_INTERFACE = 'org.freedesktop.sssd.infopipe.Domains.Domain'
+    INFOPIPE_INTERFACE = 'org.freedesktop.sssd.infopipe'
+    
+    def __init__(self):
+        self.bus = dbus.SystemBus()
+        self.sssd_obj = self.bus.get_object(self.BUS_NAME, '/org/freedesktop/sssd/infopipe')
+        self.infopipe_iface = dbus.Interface(self.sssd_obj, dbus_interface=self.INFOPIPE_INTERFACE)
+    
+    def _get_domain_path(self, domain):
+        """Convert domain name to D-Bus path format"""
+        return "/org/freedesktop/sssd/infopipe/Domains/%s" % domain.replace('.', '_2e')
```

Comment:
Please use f-strings
```suggestion
        return f"/org/freedesktop/sssd/infopipe/Domains/{domain.replace('.', '_2e')}"
```